### PR TITLE
Duck stays when player moves

### PR DESCRIPTION
**Pull request by JeromeTiong**
@ivantjh  @chesterhow

### CODE REVIEW

File: src/duck.cpp [hunk 1]
```diff
@@ -2,24 +2,80 @@
...
+	if (input->isKeyDown(UP_KEY) && !(*keysPressed)["UP"]) {
+		if (player.isValidMove(levelGrid, UP)) {
+			printf("GET: %f SET: %f", this->getX(), this->getX() + TILE_SIZE * SCALE); 
+			this->setY(this->getY() + TILE_SIZE * SCALE);
+		}
+	}
+
+	if (input->isKeyDown(DOWN_KEY) && !(*keysPressed)["DOWN"]) {
+		if (player.isValidMove(levelGrid, DOWN)) {
+			printf("GET: %f SET: %f", this->getX(), this->getX() + TILE_SIZE * SCALE);
```
Comment (chesterhow) [inline src/duck.cpp]: actually we should remove the printf stuffs.

File: src/player.cpp [hunk 2]
```diff
@@ -1,6 +1,8 @@
 #include "player.h"
 
 Player::Player() : Entity() {
+	/*this->setStartFrame(0);
+	this->setEndFrame(2);*/
```
Comment (chesterhow) [inline src/player.cpp]: Can remove this too

**Comment by chesterhow** [comment]
@ivantjh ideally we should also add a currentPosition variable to Entity

File: src/image.h [hunk 3]
```diff
@@ -152,9 +152,13 @@ class Image {
 	virtual void setColorFilter(COLOR_ARGB color) { colorFilter = color; }
 
 	// Set TextureManager
-	virtual void setTextureManager(TextureManager *textureM) {
-		textureManager = textureM;
-	}
+	virtual void setTextureManager(TextureManager *textureM) { textureManager = textureM; }
+
+	//Set StartFrame
+	virtual void setStartFrame(int startframe) { startFrame = startframe; }
```
Comment (ivantjh) [inline src/image.h]: can just use the `setFrames` func of the image class

**Comment by ivantjh** [comment]
Current Position for the coordinates of the grid?

**Comment by chesterhow** [comment]
yea coordinates. So that the player can detect theres an enemy in that spot

**Comment by ivantjh** [comment]
Not working fully, do not merge this. I fixed it on my branch.

**Pull request merged**